Assets/Scripts/GranadeerScript.cs

```diff
@@ -53,6 +53,11 @@ private void Update()
                     {
                         state = State.Attacking;
                     }
+
```

Comment:
Since the enemy can die at any state then move this outside of the switch and move it to the base class